app/scripts/services/dimItemService.factory.js

```diff
@@ -381,102 +373,215 @@
+      }
+
       if (item.owner === store.id) {
         return $q.resolve(true);
       }
 
       var stores = dimStoreService.getStores();
-      // How much space will be needed in the various stores in order to make the transfer?
+
+      // How much space will be needed (in amount, not stacks) in the target store in order to make the transfer?
       var storeReservations = {};
-      stores.forEach(function(s) {
-        storeReservations[s.id] = (s.id === store.id ? slotsNeededForTransferTo(s, item) : 0);
-      });
+      storeReservations[store.id] = item.amount;
+
       // guardian-to-guardian transfer will also need space in the vault
       if (item.owner !== 'vault' && !store.isVault) {
-        storeReservations.vault = slotsNeededForTransferTo(dimStoreService.getVault(), item);
+        storeReservations.vault = item.amount;
       }
 
-      // How many moves are needed from each
+      // How many moves (in amount, not stacks) are needed from each
       var movesNeeded = {};
-      dimStoreService.getStores().forEach(function(s) {
-        movesNeeded[s.id] = Math.max(0, storeReservations[s.id] - s.spaceLeftForItem(item));
+      stores.forEach(function(s) {
+        if (storeReservations[s.id]) {
+          movesNeeded[s.id] = Math.max(0, storeReservations[s.id] - spaceLeftWithReservations(s, item));
+        }
       });
 
       if (!_.any(movesNeeded)) {
         return $q.resolve(true);
       } else if (store.isVault || triedFallback) {
+        // TODO: pass along the move context even more, and keep track of the original item and store
+
         // Move aside one of the items that's in the way
         var moveContext = {
-          reservations: storeReservations,
           originalItemType: item.type,
-          excludes: excludes || [],
+          excludes: excludes,
           spaceLeft: function(s, i) {
-            var left = s.spaceLeftForItem(i);
+            let left = spaceLeftWithReservations(s, i);
             if (i.type === this.originalItemType) {
-              left = left - this.reservations[s.id];
+              left = left - storeReservations[s.id];
             }
             return Math.max(0, left);
           }
         };
 
         // Move starting from the vault (which is always last)
-        var move = _.pairs(movesNeeded)
+        var moves = _.pairs(movesNeeded)
               .reverse()
-              .find(function(p) { return p[1] > 0; });
-        var source = dimStoreService.getStore(move[0]);
-        var moveAsideItem = chooseMoveAsideItem(source, item, moveContext);
-        var target = chooseMoveAsideTarget(source, moveAsideItem, moveContext);
-
-        if (!target || (!target.isVault && target.spaceLeftForItem(moveAsideItem) <= 0)) {
-          return $q.reject(new Error('There are too many \'' + (target.isVault ? moveAsideItem.bucket.sort : moveAsideItem.type) + '\' items in the ' + target.name + '.'));
+              .find(([_, moveAmount]) => moveAmount > 0);
```

Comment:
what is this? why is the underscore being passed in as an element in that array?